### PR TITLE
fix(scripts): remove 2>&1 from gh api capture to prevent stderr corrupting jq input

### LINKED ISSUE #1122: Follow-up from #911: 2 item(s) (core, safety)
_Consolidated follow-up from implementation of #911._

Each section below lists scope-checked follow-up items discovered during implementation. Items are restricted to core library defects, security, safety hazards, or critical functional bugs.

### choose_merge_flag.sh:21 mixes gh stderr into jq input

scripts/choose_merge_flag.sh:21 captures `gh api repos/${repo} 2>&1` into $raw, then pipes $raw into `jq -r` (line 26). When `gh` writes any stderr noise (deprecation warning, update banner, rate-limit notice, or `GH_DEBUG=1` traces) alongside a valid JSON body on stdout, the merged blob is no longer valid JSON; `jq -r` returns empty, and the function falls through to `"target repo ${repo} allows no merge methods"` (exit 1) even though all three flags would have been valid. Repro: `gh() { echo 'warning: x' >&2; echo '{"allow_rebase_merge":true,"allow_squash_merge":true,"allow_merge_commit":true}'; }; export -f gh; . scripts/choose_merge_flag.sh; choose_merge_flag o/r` yields empty + exit 1 instead of `--rebase`. Impact: on ProjectHephaestus the `--squash` skill-level fallback masks it; on a rebase-only target repo, callers either fail or arm the wrong merge method. Fix: capture stdout-only — `if ! raw=$(gh api "repos/${repo}"); then ...` and keep a separate `2>/dev/null` or a captured-stderr variable for the error message. Add a regression test using the same gh-mock that writes to stderr.

### Integration tests skip choose_merge_flag positive paths

tests/integration/test_choose_merge_flag_sh.py only covers function definition (line 8) and missing-arg exit 2 (line 19). The three behaviors documented in scripts/choose_merge_flag.sh:9-13 — preference order rebase→squash→merge, `no methods allowed` exit 1, successful flag emission — are never exercised. The PR-911 plan's Verification section sketched a `bash -c 'gh() { case "$1" in api) echo {...};; esac; }; export -f gh; . scripts/choose_merge_flag.sh && choose_merge_flag o/r'` pattern that wasn't checked in. Add three integration tests using that mock-gh approach: (a) all-three-allowed yields `--rebase`, (b) only squash allowed yields `--squash`, (c) none allowed yields exit 1 with stderr containing `allows no merge methods`. These cover code I just wrote, so they qualify as `core` per the test-scope rule.

---

_The implementer also considered the items below and rejected them as out of scope; they are recorded in the PR body, not filed as separate issues._

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary

- `scripts/choose_merge_flag.sh:21` captured `gh api` stderr into `$raw` via `2>&1`, causing any non-JSON stderr noise (deprecation banners, rate-limit notices, `GH_DEBUG` traces) to corrupt the `jq` input and produce a spurious "allows no merge methods" exit-1 even when all methods are valid.
- Fix: drop `2>&1` so `gh` stdout is captured cleanly; `gh` writes its own error to stderr on failure, which flows through naturally.
- Add four integration tests covering the three previously-untested positive/negative paths plus the direct stderr-noise regression case.

## Test plan

- [ ] `pixi run pytest tests/integration/test_choose_merge_flag_sh.py -v --no-cov` → 6/6 pass
- [ ] `test_stderr_noise_does_not_corrupt_jq_input` is the direct regression test for the safety fix
- [ ] `test_all_methods_allowed_prefers_rebase` / `test_only_squash_allowed_yields_squash` / `test_no_methods_allowed_exits_1_with_message` cover the three core paths

Closes #1122

### CODE REVIEW

File: tests/integration/test_choose_merge_flag_sh.py [hunk 1]
```diff
@@ -25,3 +25,66 @@ def test_shell_helper_errors_on_missing_arg() -> None:
...
+def test_all_methods_allowed_prefers_rebase() -> None:
+    """Preference order: rebase wins when all three methods are enabled."""
+    json_body = '{"allow_rebase_merge":true,"allow_squash_merge":true,"allow_merge_commit":true}'
+    script = f"""
+gh() {{ case "$1" in api) printf '%s\\n' '{json_body}';; esac; }}
+export -f gh
+. {SNIPPET}
+choose_merge_flag owner/repo
+"""
+    out = subprocess.run(["bash", "-c", script], capture_output=True, text=True)
```
Comment (mvillmow) [inline tests/integration/test_choose_merge_flag_sh.py]: The four new tests repeat an identical ~5-line subprocess scaffold (build f-string with inline gh() + export -f gh + source SNIPPET + run via bash -c). Consider extracting a helper like `_run_choose(json_body, *, stderr_noise=False) -> subprocess.CompletedProcess` so each test asserts only on returncode/stdout/stderr. Minor — the explicit form is readable and matches the existing two tests, so this is a judgment call, not a blocker.

**Review by mvillmow — COMMENTED** [review]
Correct, complete fix: 2>&1 removed (gh stderr no longer corrupts jq) + 4 path/regression tests. One minor DRY nit on the test scaffold.

**Review by mvillmow — COMMENTED** [review]
LGTM — 2>&1 removed (gh stderr no longer corrupts jq) + 4 path/regression tests. Open minor test-scaffold DRY nit is a non-blocking judgment call.

**Review by mvillmow — COMMENTED** [review]
LGTM — 2>&1 removed (gh stderr no longer corrupts jq) + 4 path/regression tests cover both issue items. Open minor test-scaffold DRY nit is a non-blocking judgment call.

File: tests/integration/test_choose_merge_flag_sh.py (hunk 1 above)
Comment (mvillmow) [reply tests/integration/test_choose_merge_flag_sh.py]: WONT-FIX: intentional design — The four new tests deliberately mirror the two pre-existing tests in the same file (test_shell_helper_defines_function L8, test_shell_helper_errors_on_missing_arg L19), which also inline their own subprocess.run scaffold rather than sharing a helper. Keeping the explicit form is a documented style choice consistent with the file's existing convention; the original comment itself states 'the explicit form is readable and matches the existing two tests, so this is a judgment call, not a blocker.' Extracting _run_choose is optional, not required.

**Review by mvillmow — COMMENTED** [review]
LGTM — 2>&1 removed (gh stderr no longer corrupts jq) + 4 path/regression tests cover both issue items. Prior DRY nit accepted as deliberate style matching existing tests.
